apps/crud/src/app/todo-item.component.ts

```diff
@@ -0,0 +1,53 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  Output,
+} from '@angular/core';
+import { CommonModule, NgIf } from '@angular/common';
+import { Todo } from './todo.model';
+
+@Component({
+  selector: 'app-todo-item',
+  standalone: true,
+  imports: [CommonModule, NgIf],
+  template: `<p>{{ todo.title }}</p>
+    <button
+      [disabled]="loading && todo.id !== selectedId"
+      (click)="update(todo)">
+      Update
+    </button>
+    <button
+      [disabled]="loading && todo.id !== selectedId"
+      (click)="remove(todo)">
+      Delete
+    </button>
+    <ng-container *ngIf="loading && todo.id === selectedId">{{
+      status
+    }}</ng-container> `,
+  styles: [],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class TodoItemComponent {
+  @Input() todo!: Todo;
+  @Input() loading!: boolean;
+  @Output() deleteTodoEvent = new EventEmitter<number>();
+  @Output() updateTodoEvent = new EventEmitter<Todo>();
+
+  selectedId = 0;
+
+  status = '';
```

Comment:
this is not very reactive programming. A better solution will to create a local store to this component which will handle the loading and update and delete of ONE todo. This way, each line will have its own instance of a store, and you will not need to have this `selectedId` that is a big ugly, IMO.

Reply:
okay I've made the changes.. please you can check it out and lemme have your thoughts

Reply:
yes that's perfect, isn't it nicer ?

Reply:
Yeah ...thanks very much